### DISCUSSION TITLE
Umi项目中，package.json中只会引入一个`umi`的dependencies是吗，其他的都在umi的依赖里面，我们不用管。这个就是【依赖收敛】没错吧？

### DISCUSSION THREAD

**Discussion by liaodalin19903** [umijs/umi#10422] [Q&A] [answered]
在阅读Umi [package.json](https://umijs.org/docs/guides/directory-structure#packagejson)的时候，

> 与 Umi 3 不同，Umi 4 不会自动注册 package.json 中以 @umijs/preset-、@umijs/plugin-、umi-preset- 和 umi-plugin- 开头的插件、预设，若你需要自定义额外的插件、预设，需要手动配置到 [plugins](https://umijs.org/docs/api/config#plugins) 。

请问下，
1.自动注册的注册是什么意思？
2.插件和预设有什么区别呢？
3.Umi项目中，package.json中只会引入一个`umi`的dependencies是吗，其他的都在umi的依赖里面，我们不用管。这个就是【依赖收敛】没错吧？

**Comment by xiaohuoni** [comment, marked as answer]
1.自动注册的注册指的是插件的注册，也就是对应手动注册应该使用 `plugins:['some']`
2.这里的预设指的是插件集 就是 preset ，顾名思义就是插件集合，把几个插件捆绑在一起，同时提供给用户。
3.不仅仅是【依赖收敛】，还有更好的“锁定版本”和 “import all from umi” 等
